File: src/main/java/me/miki/shindo/management/cosmetic/cape/impl/Cape.java
package me.miki.shindo.management.cosmetic.cape.impl;

import me.miki.shindo.api.roles.Role;
import me.miki.shindo.management.cosmetic.cape.CapeCategory;
import me.miki.shindo.utils.animation.simple.SimpleAnimation;
import net.minecraft.util.ResourceLocation;

public class Cape {

    private final String name;
    private final ResourceLocation cape;
    private final CapeCategory category;
    private final Role requiredRole;

    private final SimpleAnimation animation = new SimpleAnimation();


    public String getName() {
        return name;
    }

    public ResourceLocation getCape() {
        return cape;
    }

    public CapeCategory getCategory() {
        return category;
    }

    public Role getRequiredRole() {
        return requiredRole;
    }

    public SimpleAnimation getAnimation() {
        return animation;
    }

    public Cape(String name, ResourceLocation cape, CapeCategory category, Role requiredRole) {
        this.name = name;
        this.category = category;
        this.cape = cape;
        this.requiredRole = requiredRole;
    }

}

File: src/main/kotlin/com/shindoclient/shindo/management/cosmetic/cape/impl/Cape.kt
package com.shindoclient.shindo.management.cosmetic.cape.impl

import com.shindoclient.shindo.api.roles.Role
import com.shindoclient.shindo.management.cosmetic.cape.CapeCategory
import com.shindoclient.shindo.ui.animation.v2.value.SimpleAnimation
import net.minecraft.util.ResourceLocation

open class Cape(
    private val name: String,
    private val cape: ResourceLocation?,
    private val category: CapeCategory,
    private val requiredRole: Role,
) {
    private val animation = SimpleAnimation()

    fun getName(): String = name

    fun getCape(): ResourceLocation? = cape

    fun getCategory(): CapeCategory = category

    fun getRequiredRole(): Role = requiredRole

    fun getAnimation(): SimpleAnimation = animation
}
